File: OsmAnd/src/net/osmand/plus/card/color/palette/IColorsPalette.java
package net.osmand.plus.card.color.palette;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import net.osmand.plus.card.color.palette.data.PaletteColor;

public interface IColorsPalette {
	void updatePaletteColors(@Nullable PaletteColor targetPaletteColor);
	void updatePaletteSelection(@Nullable PaletteColor oldColor, @NonNull PaletteColor newColor);
}

File: OsmAnd/src/net/osmand/plus/card/color/palette/main/v2/IColorsPalette.kt
package net.osmand.plus.card.color.palette.main.v2

import net.osmand.shared.palette.domain.PaletteItem

// TODO: IPaletteView
interface IColorsPalette {
	fun updatePaletteItems(targetItem: PaletteItem?)
	fun updatePaletteSelection(oldItem: PaletteItem?, newItem: PaletteItem)
}